R/moma_arguments.R

```diff
@@ -320,24 +187,32 @@ cluster <- function(..., w = NULL, ADMM = FALSE,
         acc = acc, prox_eps = eps,
         P = "UNORDEREDFUSION"
     )
-    class(arglist) <- "moma_sparsity"
+    class(arglist) <- "_moma_sparsity_type"
     return(arglist)
 }
 
 
-#' Algorithm settings for solving a penalzied SVD problem
+#' Algorithm settings for solving the penalized SVD problem
+#'
+#' This function is used to specify the \code{pg_settings} argument
+#' in the \code{moma_*pca}, \code{moma_*cca}, and \code{moma_*ldaa}
+#' types of functions.
 #'
 #' To find an (approximate) solution to a penalized SVD (Singular Value Decomposition) problem is to solve two
 #' penalized regression problems iteratively (outer loop). Each penalized regression (inner loop)
 #' is solved using one of the three algorithms: ISTA (Iterative Shrinkage-Thresholding Algorithm),
 #' FISTA (Fast Iterative Shrinkage-Thresholding Algorithm) and
 #' One-step ISTA (an approximated version of ISTA).
-#' @param ... to force users to specify arguments by names
-#' @param EPS precision for outer loop
-#' @param MAX_ITER the maximum number of iterations for outer loop
-#' @param EPS_inner precision for inner loop
-#' @param MAX_ITER_inner the maximum number of iterations for inner loop
-#' @param solver a string in \code{c("ista", "fista", "onestepista")}.
+#' @param ... To force users to specify arguments by names.
+#' @param EPS Precision for outer loop.
+#' @param MAX_ITER The maximum number of iterations for outer loop.
+#' @param EPS_inner Precision for inner loop.
+#' @param MAX_ITER_inner The maximum number of iterations for inner loop.
+#' @param solver A string in \code{c("ista", "fista", "onestepista")}, representing ISTA (Iterative Shrinkage-Thresholding Algorithm),
+#'              FISTA (Fast
+#'              Iterative Shrinkage-Thresholding Algorithm) and One-step ISTA (an approximated
+#'              version of ISTA) respectively.
```

Comment:
Nothing to fix, but interesting connection: it turns out the One-Step ISTA is not just an approximation of ISTA, but rather an instance of the Alternating Manifold Proximal Gradient method of Chen *et al.* (https://arxiv.org/abs/1903.11576). (The manifold stuff is hidden in the normalization step, which is really projection / retraction onto the unit sphere manifold.) 

As I argue in my Manifold SFPCA paper, this is sometimes actually not a bad approach, but it's hard to give theoretical guidance as to what's best in general.